api/envoy/extensions/filters/http/ext_authz/v3/ext_authz.proto

```diff
@@ -215,6 +215,29 @@ message ExtAuthz {
   // Whether to increment cluster statistics (e.g. cluster.<cluster_name>.upstream_rq_*) on authorization failure.
   // Defaults to true.
   google.protobuf.BoolValue charge_cluster_response_stats = 20;
+
+  // Whether to encode the raw headers (i.e. unsanitized values & unconcatenated multi-line headers)
+  // in authentication request.
+  //
+  // The string data type cannot contain non-UTF-8-compliant data or it will create an error when
+  // decoding. Thus, if this field is false, header values will be sanitized, and any non-UTF-8-
+  // compliant bytes will be replaced with '!'.
+  //
+  // When this is set to true, header values are not sanitized. Headers will be added to
+  // :ref:`headers_bytes<envoy_v3_api_field_service.auth.v3.AttributeContext.HttpRequest.header_map>`
+  // instead of :ref:`headers<envoy_v3_api_field_service.auth.v3.AttributeContext.HttpRequest.headers>`.
+  //
+  // When this is set to true, headers with the same key will also not be combined into a single
+  // comma separated header.
+  //
+  // If set to true with an
```

Comment:
From comment readability perspective, I think it would be better to restructure this as:
// If this is true .... (include the implications).
//
// If this is false ... (include the implications).
// 
// Recommendation [as you have at the moment].

Reply:
Done.